### PR TITLE
chore(ci): build arm64 docker image

### PR DESCRIPTION
**Pull request by huangjw806**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

As the title.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

File: ci/scripts/docker.sh [hunk 1]
```diff
@@ -17,23 +17,23 @@ components=(
 for component in "${components[@]}"
 do
   echo "--- docker build and tag : ${component}"
-  docker build -f docker/Dockerfile -t "${ghcraddr}/${component}:latest" --target "${component}" .
+  docker build -f docker/Dockerfile -t "${ghcraddr}/${component}:latest-$(uname -m)" --target "${component}" .
```
Comment (skyzh) [inline ci/scripts/docker.sh:20]: We only need to `uname -m` once and save it in a variable.

File: ci/scripts/multi-arch-docker.sh [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  "compute-node"
+  "meta-node"
+  "frontend-node"
+  "compactor-node"
+)
+
+echo "--- ghcr login"
+echo "$GHCR_TOKEN" | docker login ghcr.io -u "$GHCR_USERNAME" --password-stdin
+
+for component in "${components[@]}"
```
Comment (skyzh) [inline ci/scripts/multi-arch-docker.sh:20]: Would be better if we can merge the two scripts into one. e.g.

```
docker.sh --merge
docker.sh --build
```

... if you are a bash master 🤣

File: ci/workflows/docker.yml [hunk 3]
```diff
@@ -31,9 +25,10 @@ steps:
       PUSH_GHCR: false
     retry: *auto-retry
 
-  - label: "docker-build-push-main"
+  - label: "docker-build-push: amd64"
```
Comment (skyzh) [inline ci/workflows/docker.yml:28]: Can we use build matrix here?

**Review by skyzh — APPROVED** [review]
Generally LGTM

File: ci/workflows/docker.yml (hunk 3 above)
Comment (huangjw806) [reply ci/workflows/docker.yml:28]: > 

Agent queue cannot use matrix.

**Pull request merged**